### DISCUSSION TITLE
ImGui aborting program, giving "Mismatched version string!" error.

### DISCUSSION THREAD

**Discussion by Wyndfall** [ocornut/imgui#4622] [New Users Build/Link/Run issues ONLY!]
Hi, I am trying to add imgui windows to a C++ project exported from SHADERed.
My code:
https://pastebin.com/AcFDyUhr
And my CMakeLists:
https://pastebin.com/eKHtipmy
The project builds and opens a window, but gives a Debug Error saying abort() has been called, and gives the error:

Assertion failed: strcmp(version, "1.85 WIP") == 0 && "Mismatched version string!", file C:\dev\vcpkg\packages\imgui_x86-windows\imgui.cpp, line 7254

In the console.
I'd really appreciate any help fixing this, thanks.

**Comment by PathogenDavid** [comment]
That error typically indicates you're using one version for the ImGui headers and building against a different one. (IE: You're including the Dear ImGui 1.84 headers but you're building against 1.85 WIP.) If you check the value of `IMGUI_VERSION` you'll find it's something other than 11.85 WIP1.

Looking at your CMake script, it looks like you've manually hard-coded building the source files from `C:/dev/vcpkg/packages/imgui_x86-windows/` but you're also using whatever `find_package(IMGUI REQUIRED)` found. Kinda surprised you're not getting an error, but that's probably your root cause.

**Reply by Wyndfall** [reply]
Yep, that was it! I just removed everything that vcpkg did with imgui and did it all manually to fix it- along with fixing a few bugs- and now its working great.
Thanks so much.
